### PR TITLE
[test]: PB-81 인증/인가 최소 통합테스트 + refresh 400 계약 정합화

### LINKED ISSUE #172: [test] PB-81 인증/인가 최소 통합테스트
### 무엇을 테스트하나?
- 인증/인가 최소 통합테스트 (AuthController + SecurityFilterChain + JwtAuthenticationFilter 동작)

### 왜 필요한가? (버그 방지/신뢰성/회귀 대응)
- 인증/인가는 보안 핵심이며 필터 체인/예외 매핑이 잘못되면 전체 서비스 접근이 막힘
- 단위테스트로는 필터 체인, 인증 진입점(401), 실제 엔드포인트 보호를 검증할 수 없음

- ❌ 전체 인증 플로우 E2E
- ❌ JWT 만료/서명 변형 등 모든 edge 케이스

### 배경
- AuthService 단위 테스트는 추가 완료됨
- 통합 구간(필터 체인/인증 진입점/보호 API 접근)은 미검증

---

<details><summary>테스트 전체 설계 (선택사항)</summary>

### 대상 모듈/기능
- SecurityConfig, JwtAuthenticationFilter, AuthController

### 테스트 범위
- [ ] Unit (클래스/메서드 단위)
- [x] Integration (레이어 간 통합)
- [ ] E2E (전체 흐름)
- [ ] UI/UX (화면/인터랙션)
- [ ] Performance (성능/부하)
- 기타:

### 테스트 전략/접근법
- 보호 API에 대한 접근 제어(401/200)를 중심으로 최소 2~3개 통합 시나리오 검증

### 전체 시나리오 (상세)
1. 정상 케이스: 로그인 성공 → 보호 API 200
2. 예외 케이스: 보호 API 토큰 없음 → 401
3. 경계값 케이스:  refresh 실패(잘못된 토큰) → 400/401
4. 동시성/경쟁 조건 (필요 시): 없음

</details>

---

## PR 분할 전략 (테스트 파일 많을 시)
- [x] Single PR (전체 테스트 한 번에)
- [ ] Stacked PR (레이어/모듈별 분할)

### PR 체인 계획 (Stacked PR 시)
1. PR #1: [Unit 테스트] - XX개 케이스
2. PR #2: [Integration 테스트] - XX개 케이스
3. PR #3: [E2E 테스트] - XX개 케이스

### 부분 머지 안전성
- PR1만 머지: ✅ 안전 (Unit 커버리지 증가)

---

<details><summary>품질 목표 (선택사항)</summary>

### 커버리지
- 현재: 0%
- 목표: 최소 통합 시나리오 3개 보장
- 변화: +3 시나리오

### 안정성
- [ ] 플래키 테스트 없음
- [ ] 플래키 있음 (원인/대응: )

</details>

---

## 영향 범위
- [ ] ⚠️ Breaking change (무엇이 깨지는지 / 마이그레이션 전략: )
- [x] 런타임 코드 변경 없음 (테스트만)
- [ ] 런타임 코드 변경 있음 (요약: )
- [ ] DB 스키마/마이그레이션 변경
- [ ] CI/빌드 파이프라인 변경
- [ ] 테스트 환경/설정 변경

---

### 리스크
- 테스트 실행 시간 증가: 낮음
- 외부 의존성 필요: Docker/Testcontainers

### 롤백
- [x] 단순 revert로 가능
- [ ] 테스트 비활성/스킵 필요

---

## 연관 이슈/PR
- 기능 이슈: 없음
- 선행 PR: AuthService 단위 테스트 PR

---

### TC1: Security/Auth - 보호 API 토큰 없음
목적: 인증 진입점 동작 확인

세부 케이스:
- 정상 케이스: 없음
- 예외 케이스: 토큰 없이 /api/auth/me 접근 → 401
- 경계값 케이스: 없음

### TC2: Auth - 로그인 성공 후 보호 API 접근
목적: 로그인 → 토큰 발급 → 인증 필터 통과 확인

세부 케이스:
- 정상 케이스: /api/auth/login 성공 후 /api/auth/me 200
- 예외 케이스: 없음
- 경계값 케이스: 없음

### TC3: Auth - refresh 실패
목적: refresh 경로의 인증/예외 매핑 확인

세부 케이스:
- 정상 케이스: 없음
- 예외 케이스: invalid refresh token → 400/401
- 경계값 케이스: 없음

---

<details><summary>증빙/참고 자료 (선택사항)</summary>

### 가이드 /문서
- 테스트 가이드:
- 관련 문서:

### 테스트 완료 후 (전체 PR 머지 시)
- 테스트 리포트:
- 커버리지 리포트:
- 실행 결과 로그:

</details>

### PR DESCRIPTION
**Pull request by PHJ2000**
# 🧪 Test PR (Updated): PB-81

## 🔗 이슈 링크 (필수)
- Jira: https://parkjaehong.atlassian.net/browse/PB-81
- GitHub: Related #172
- GitHub: Closes #172 (Final PR만)

> PR 제목(권장): `test: PB-81 인증/인가 최소 통합테스트 + refresh 400 계약 정합화`

---

### 전체 중 위치
- 전체 이슈 PB-81의 1/1 단계 (Single PR)
- 선행 PR: `test: PB-80 AuthService 단위 테스트 추가` #171 (✅ merge됨)

### 변경/추가 상세
- [x] **신규 테스트 추가**
  - Integration: `AuthSecurityIntegrationTest` 내 3개 TC (TC1~TC3)
- [x] **기존 테스트 수정**
  - Unit: `AuthServiceTest`의 refresh 실패 케이스 기대값을 `ApiException(400, INVALID_REQUEST)`로 정합화
  - 이유: refresh 실패를 400으로 통일하는 런타임 정책과 단위테스트 기대 예외 타입이 불일치하여 CI 실패 발생
- [x] **문서 추가/갱신**
  - `docs/troubleshooting/2026-02-05-ci-auth-refresh-unit-test-exception-mismatch.md` (원인 추적/해결 기록)
  - `docs/troubleshooting/docs-README.md` 인덱스 업데이트
- [x] **DevX(훅) 개선 포함**
  - `.githooks/*`: 브랜치명이 `pb-81`처럼 소문자여도 `PB-81`로 인식/주입되도록 개선
  - `.githooks/pre-commit`: Java 파일이 staged된 경우에만 `spotlessApply` 실행 + `GRADLE_USER_HOME` 레포 내부로 fallback

---

### Covers
- **TC1**: 보호 API 토큰 없음 → `GET /api/auth/me` → `401 (UNAUTHORIZED)`
- **TC2**: 로그인 성공 후 보호 API 접근 → `POST /api/auth/login` 후 `GET /api/auth/me` → `200`
- **TC3**: refresh 실패(유효하지 않은 refresh token) → `POST /api/auth/refresh` → `400 (INVALID_REQUEST)`

### Not covered
- E2E 전체 인증 플로우 (이유: 최소 통합테스트 범위 밖)
- JWT edge case(만료/서명 변형 등) 전체 커버 (이유: 최소 통합테스트 범위 밖)
- lastLoginAt DB 반영(서비스-영속성 통합테스트) (이유: 별도 이슈로 분리 예정)

---

### 로컬
```bash
./gradlew test --tests com.beachcheck.integration.AuthSecurityIntegrationTest
./gradlew test --tests com.beachcheck.service.AuthServiceTest
```
- 결과: (작성)

### CI
- (자동 첨부)

---

## 🌐 영향 범위 체크 (필수)
- [x] 런타임 코드 변경
  - 요약: `AuthService.refresh`에서 invalid refresh token을 `ApiException(INVALID_REQUEST)`로 통일(400 계약)
- [x] 테스트 변경
  - 요약: `AuthServiceTest` refresh 실패 케이스를 400 정책에 맞게 수정 + 인증 최소 통합테스트 추가
- [x] 문서 변경
  - 요약: CI 원인 추적/해결 트러블슈팅 문서 추가 + 인덱스 갱신
- [x] 개발 도구(훅) 변경
  - 요약: JIRA 키 자동 주입(소문자 pb- 포함) + pre-commit 비용/환경 의존 개선
- [ ] DB/마이그레이션 변경

---

### 집중해서 볼 테스트/로직
- `src/test/java/com/beachcheck/integration/AuthSecurityIntegrationTest.java` (TC1~TC3)
- `src/main/java/com/beachcheck/service/AuthService.java` (refresh 실패 400 계약)
- `src/test/java/com/beachcheck/service/AuthServiceTest.java` (refresh 실패 케이스 기대값 정합화)

### (선택) 트러블슈팅 기록/DevX
- `docs/troubleshooting/2026-02-05-ci-auth-refresh-unit-test-exception-mismatch.md`
- `.githooks/prepare-commit-msg`, `.githooks/commit-msg`, `.githooks/pre-commit`

### 트레이드오프
- refresh 실패를 400으로 통일해 테스트/클라이언트 분기를 단순화하는 대신,
  기존에 상태코드(404/409 등)에 의존하던 클라이언트가 있다면 영향 가능
- 근거: OAuth 2.0 RFC 6749 §6, §5.2 (token endpoint error 기본 400)  https://www.rfc-editor.org/rfc/rfc6749

### CODE REVIEW

File: src/main/java/com/beachcheck/service/AuthService.java [hunk 1]
```diff
@@ -139,6 +137,10 @@ public TokenResponseDto refresh(String refreshTokenStr) {
     return TokenResponseDto.of(newAccessToken, jwtUtils.getAccessTokenExpiration());
   }
 
+  private ApiException invalidRefreshToken() {
```
Comment (geonusp) [inline src/main/java/com/beachcheck/service/AuthService.java]: https://www.rfc-editor.org/rfc/rfc6749#section-5.2에 따르면 
refresh token 파라미터 누락/형식 깨짐/중복은 Invalid_REQUEST라고 명시되어있고
refresh token invalid/expired/revoked/다른 클라이언트에 발급된 토큰은 Invalid_GRANT 라고 명시되어있기 때문에 유효하지 않은 리프레시 토큰은 INVALID_GRANT가 더 적절할 거 같아서 TODO로 남기거나 바로 수정부탁드립니다.

File: src/test/java/com/beachcheck/integration/AuthSecurityIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    @DisplayName("TC1: 토큰 없이 내 정보 조회 시 401")
+    void me_withoutToken_returnsUnauthorized() throws Exception {
+      // Given: 인증 헤더 없음
+
+      // When & Then: 인증 진입점에서 401 계약 응답
+      mockMvc
+          .perform(get(ME_URI))
+          .andExpect(status().isUnauthorized())
+          .andExpect(
+              ReservationTestFixtures.problemDetail(
```
Comment (geonusp) [inline src/test/java/com/beachcheck/integration/AuthSecurityIntegrationTest.java]: 예약 관련 로직이 아닌데 예약 테스트 픽스처를 쓰는 것은 오해의 여지가 있으므로 예약 테스트 픽스처에서 다른 로직에서도 쓰이는 것을 따로 모아 분리하거나 예약 테스트 픽스처 이름을 바꾸는 것을 고려하는 것이 좋아보입니다.

Comment (geonusp) [inline src/test/java/com/beachcheck/integration/AuthSecurityIntegrationTest.java]: Image
에서 
expectedTitle에도 에러코드 상수가 들어가있는데  title은 클라이언트에게 보여주는 '사람이 읽을 수 있는문구'가 들어가는게 좋아보입니다.

File: src/test/java/com/beachcheck/integration/AuthSecurityIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+          .andExpect(jsonPath("$.name").value(USER_NAME));
+    }
+  }
+
+  @Nested
+  @DisplayName("리프레시 토큰")
+  class RefreshTokenTests {
+
+    @Test
+    @DisplayName("TC3: 유효하지 않은 리프레시 토큰이면 400")
```
Comment (geonusp) [inline src/test/java/com/beachcheck/integration/AuthSecurityIntegrationTest.java]:
```java
private static final String INVALID_REFRESH_TOKEN = "invalid-refresh-token";
```
이런식으로 유효하지않은 리프레시 토큰을 적으셨는데 필드누락/빈값인지 형식 불일치/파싱 불가 인지 존재하지않음/폐기됨/만료됨인지 구분이 모호합니다. 각각의 테스트 케이스를 만들어서 테스트 하는 방식을 추천드립니다.

File: src/test/java/com/beachcheck/integration/AuthSecurityIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+  @DisplayName("보호 API 접근")
+  class ProtectedApiAccessTests {
+
+    @Test
+    @DisplayName("TC1: 토큰 없이 내 정보 조회 시 401")
+    void me_withoutToken_returnsUnauthorized() throws Exception {
+      // Given: 인증 헤더 없음
+
+      // When & Then: 인증 진입점에서 401 계약 응답
+      mockMvc
```
Comment (geonusp) [inline src/test/java/com/beachcheck/integration/AuthSecurityIntegrationTest.java]: WWW-authentication 관련 설정을 저희가 안해서 나중에 리팩토링 스프린트때 해야할거같습니다 관련해서 TODO 주석 남겨 두면 리팩토링때 편할거같습니다.

File: docs/troubleshooting/troubleshooting-ci-auth-refresh-unit-test-exception-mismatch.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
+# 2026-02-05 CI 실패: Auth refresh 단위테스트 예외 타입 불일치 (400 정책)
```
[the remark this answers is not in the document]
Comment (PHJ2000) [reply docs/troubleshooting/troubleshooting-ci-auth-refresh-unit-test-exception-mismatch.md]: 수정완료했습니다!

File: src/test/java/com/beachcheck/integration/AuthSecurityIntegrationTest.java [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.test.web.servlet.MvcResult;
+
+@DisplayName("인증/인가 최소 통합 테스트")
+class AuthSecurityIntegrationTest extends ApiTest {
+
+  private static final String LOGIN_URI = "/api/auth/login";
```
[the remark this answers is not in the document]
Comment (PHJ2000) [reply src/test/java/com/beachcheck/integration/AuthSecurityIntegrationTest.java]: 맞습니다. 기존에 만들어둔 ApiRoutes를 제가 놓쳤습니다. 해당 테스트 URI는 ApiRoutes 상수로 정리했고, JSON 요청 구간은 MediaType.APPLICATION_JSON으로 일관되게 유지했습니다.

File: src/test/java/com/beachcheck/integration/AuthSecurityIntegrationTest.java [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+  private String email;
+
+  @BeforeEach
+  void setUp() {
+    email = "auth_it_" + UUID.randomUUID().toString().substring(0, 8) + "@test.com";
+
+    User testUser = new User();
+    testUser.setEmail(email);
+    testUser.setPassword(passwordEncoder.encode(RAW_PASSWORD));
+    testUser.setName(USER_NAME);
```
[the remark this answers is not in the document]
Comment (PHJ2000) [reply src/test/java/com/beachcheck/integration/AuthSecurityIntegrationTest.java]: 추가로 동일 패턴 중복도 확인해 (1) uniqueEmail/uniqueBeachCode 생성 유틸 공통화, (2) JwtUtilsTest의 User 생성을 UserTestFixtures로 통일, (3) FavoriteTestFixtures의 beach/favorite fixture 책임 분리까지 후속 리팩토링으로 진행하겠습니다. 해당 변경은 별도 커밋으로 분리하겠습니다.

File: src/test/java/com/beachcheck/integration/AuthSecurityIntegrationTest.java (hunk 2 above)
Comment (PHJ2000) [reply src/test/java/com/beachcheck/integration/AuthSecurityIntegrationTest.java]: 좋은 코멘트 감사합니다. 공통 에러 검증 로직은 ApiErrorTestFixtures로 분리하고, 예약 전용 헬퍼만 ReservationTestFixtures에 남기도록 정리했습니다.

Comment (PHJ2000) [reply src/test/java/com/beachcheck/integration/AuthSecurityIntegrationTest.java]: 확인했습니다. 다만 한글화를 따로 진행하지 않은 이유는 src/main/java/com/beachcheck/config/SecurityConfig.java:59 해당 부분에서 타이틀을 사실상 errorcode와 동일하게 사용하고 있고, src/main/java/com/beachcheck/exception/GlobalExceptionHandler.java:34-36부분에서 title==code로 사용하고 있는 등 사용자용 문구라기 보다 코드 대체값으로 쓰는 구간이 많습니다. 이에 고쳐야 할 범위가 많으니, 한글화 부분은 프론트엔드부분에서 진행한느게 합리적이라고 생각했기 때문입니다.
